src/sycl/CMakeLists.txt

```diff
@@ -74,10 +74,20 @@ if(WITH_ROCM_BACKEND)
 endif()
 
 if(WITH_CPU_BACKEND)
+  find_package(OpenMP REQUIRED)
+  if(APPLE)
+    if(CMAKE_VERSION VERSION_LESS "3.16")
+      message(FATAL_ERROR "CMake 3.16.0+ is required for macOS OpenMP support!")
+    endif()
+  endif()
   add_library(hipSYCL_cpu SHARED ${HIPSYCL_SOURCES})
   target_compile_definitions(hipSYCL_cpu PRIVATE HIPSYCL_BOOTSTRAP=1)
   target_compile_options(hipSYCL_cpu PRIVATE --hipsycl-platform=cpu ${COMMON_SYCLCC_OPTIONS})
   target_include_directories(hipSYCL_cpu PRIVATE ${INCLUDE_DIRS} ${HIPSYCL_SOURCE_DIR}/contrib/hipCPU/include)
+  if(APPLE)
+    target_include_directories(hipSYCL_cpu PRIVATE ${OpenMP_CXX_INCLUDE_DIRS})
```

Comment:
potentially, I would like to add them to the `syclcc` compiler itself as intrinsic flags it passes to clang but did not find a quick way to achieve that.
I tried hacking the Python script but this did not change the compile error in this line.

Reply:
Yes, we would need a `syclcc` configuration setting for the CPU compilation flags as described above.

Reply:
Yes, I hard-coded this in between but was still not working, so I left it for the time being as this.